### ISSUE TITLE
Serialization: Info() and Infow() Produce Different Log Entries Given the Same Arguments

### ISSUE THREAD

**Issue by nbcarey** [uber-go/zap#954]
These two virtually identical log statements with essentially identical arguments produce different log entries. I would expect two convenience wrappers over the same basic core would produce identical log entries given the same arguments.

```go
log.Info(  "Info()",  []string{}, "name", "this-is-a-name", "product", product )
log.Infow( "Infow()",             "name", "this-is-a-name", "product", product )
```

**Zap Version:** 1.16.0

**Steps To Reproduce**

1. Run this playground: https://play.golang.org/p/kOm9CohCweK
    
    Source code can be found here: [main.go.txt](https://github.com/uber-go/zap/files/6541254/main.go.txt)
 
2. Observe the two log entries written to the console.

**Expected Results** (formatted for readability)

* `log.Info()` writes this JSON to the log:
    ```json
    {
      "level": "info",
      "ts": 1257894000,
      "caller": "sandbox719407275/prog.go:44",
      "msg": "Info()",
      "name": "this-is-a-name",
      "product": {
        "Id": 123,
        "StyleNumber": "style-number",
        "StyleGroup": "style-group",
        "Name": "product name",
        "Brand": "brand name",
        "Price": {
          "CC": "FR",
          "Value": {
            "Units": 987,
            "Cents": 65
          }
        }
      }
    }
    ```

* `log.Infow()` writes this JSON to the log:
    ```json
    {
      "level": "info",
      "ts": 1257894000,
      "caller": "sandbox719407275/prog.go:45",
      "msg": "Infow()",
      "name": "this-is-a-name",
      "product": {
        "Id": 123,
        "StyleNumber": "style-number",
        "StyleGroup": "style-group",
        "Name": "product name",
        "Brand": "brand name",
        "Price": {
          "CC": "FR",
          "Value": {
            "Units": 987,
            "Cents": 65
          }
        }
      }
    }
    ```

**Actual Results** (formatted for legibility)

* `log.Info()` writes this JSON to the log:
    ```json
    {
      "level": "info",
      "ts": 1257894000,
      "caller": "sandbox719407275/prog.go:44",
      "msg": "Info()[]namethis-is-a-nameproduct{123 style-number style-group product name brand name {FR {987 65}}}"
    }
    ```

* `log.Infow()` writes this JSON to the log:
    ```json
    {
      "level": "info",
      "ts": 1257894000,
      "caller": "sandbox719407275/prog.go:45",
      "msg": "Infow()",
      "name": "this-is-a-name",
      "product": {
        "Id": 123,
        "StyleNumber": "style-number",
        "StyleGroup": "style-group",
        "Name": "product name",
        "Brand": "brand name",
        "Price": {
          "CC": "FR",
          "Value": {
            "Units": 987,
            "Cents": 65
          }
        }
      }
    }
    ```

**Comment by moisesvega** [comment]
This behavior is by design; If we look at the documentation for [SugarLogger](https://pkg.go.dev/go.uber.org/zap#SugaredLogger) we can read the following.

```
A SugaredLogger wraps the base Logger functionality in a slower, but less verbose, API. 
Any Logger can be converted to a SugaredLogger with its Sugar method.

Unlike the Logger, the SugaredLogger doesn't insist on structured logging. 
For each log level, it exposes three methods: one for loosely-typed structured logging, 
one for println-style formatting, and one for printf-style formatting.
For example, SugaredLoggers can produce InfoLevel output
with Infow ("info with" structured context), Info, or Infof.
```

And if we look at the implementation for [Infow](https://github.com/uber-go/zap/blob/v1.17.0/sugar.go#L178) and [info](https://github.com/uber-go/zap/blob/v1.17.0/sugar.go#L101)

```
// Infow logs a message with some additional context. The variadic key-value
// pairs are treated as they are in With.
func (s *SugaredLogger) Infow(msg string, keysAndValues ...interface{}) {...}

// Info uses fmt.Sprint to construct and log a message.
func (s *SugaredLogger) Info(args ...interface{}) {...}
```
While zap.SugaredLogger supports an Info method, the info method does not do structured logging, it uses `fmt.Sprint`. Meaning you will get a concatenated string in the `msg`

**Comment by nbcarey** [comment]
> This behavior is by design; If we look at the documentation for [SugarLogger](https://pkg.go.dev/go.uber.org/zap#SugaredLogger) we can read the following.
> 
> ```
> A SugaredLogger wraps the base Logger functionality in a slower, but less verbose, API. 
> Any Logger can be converted to a SugaredLogger with its Sugar method.
> 
> Unlike the Logger, the SugaredLogger doesn't insist on structured logging. 
> For each log level, it exposes three methods: one for loosely-typed structured logging, 
> one for println-style formatting, and one for printf-style formatting.
> For example, SugaredLoggers can produce InfoLevel output
> with Infow ("info with" structured context), Info, or Infof.
> ```
> 
> And if we look at the implementation for [Infow](https://github.com/uber-go/zap/blob/v1.17.0/sugar.go#L178) and [info](https://github.com/uber-go/zap/blob/v1.17.0/sugar.go#L101)
> 
> ```
> // Infow logs a message with some additional context. The variadic key-value
> // pairs are treated as they are in With.
> func (s *SugaredLogger) Infow(msg string, keysAndValues ...interface{}) {...}
> 
> // Info uses fmt.Sprint to construct and log a message.
> func (s *SugaredLogger) Info(args ...interface{}) {...}
> ```
> 
> While zap.SugaredLogger supports an Info method, the info method does not do structured logging, it uses `fmt.Sprint`. Meaning you will get a concatenated string in the `msg`

And that is exactly my point. It is inconsistent behavior that would be unexpected by _**any**_ user of the tooling.

**Comment by abhinav** [comment]
Hey @nbcarey, thanks for the report. This behavior of the SugaredLogger was chosen to ease migration for users coming from systems like [logrus](https://pkg.go.dev/github.com/sirupsen/logrus) where `Info`/`Error`/etc. behave similar to `fmt.Sprint`. Easing this migration was a priority when we developed Zap.

While we agree that this can sometimes be a source of confusion, we unfortunately cannot change this behavior. We are currently opposed to making breaking changes to Zap that would necessitate a v2. 

Sorry we don't have a good solution for the confusion caused by this behavioral difference at this time. Closing.
